db/snapshot-ctl.cc

```diff
@@ -101,7 +101,8 @@ future<> snapshot_ctl::do_take_column_family_snapshot(sstring ks_name, std::vect
             throw std::invalid_argument("Do not take a snapshot of a materialized view or a secondary index by itself. Run snapshot on the base table instead.");
         }
     }
-    co_await replica::database::snapshot_tables_on_all_shards(_db, ks_name, std::move(tables), std::move(tag), bool(sf));
+    bool snap_views = (av == allow_view_snapshots::no);
```

Comment:
Is this flipped? allow_view_snaphots::no -> snap_views = true.

Reply:
with allow_view_snapshots::yes the caller (truncate/drop) is snapping the views on its own, hence we don't need to auto-snap them.  Maybe we can push this functionality here, and get rid of allow_view_snapshots.